src/search/landmarks/landmark_graph.h

```diff
@@ -55,104 +57,83 @@ class LandmarkNode {
+    // needed by both landmarkgraph-factories and non-landmarkgraph-factories
+    const Nodes &get_nodes() const {
         return nodes;
     }
-
-    inline int number_of_landmarks() const {
+    // needed by both landmarkgraph-factories and non-landmarkgraph-factories
+    int get_num_landmarks() const {
         return nodes.size();
     }
-
-    // ------------------------------------------------------------------------------
-    // methods needed only by landmarkgraph-factories
-    explicit LandmarkGraph(const TaskProxy &task_proxy);
-
-    inline LandmarkNode &get_simple_lm_node(const FactPair &a) const {
-        assert(simple_landmark_exists(a));
-        return *(simple_lms_to_nodes.find(a)->second);
-    }
-    inline LandmarkNode &get_disj_lm_node(const FactPair &a) const {
-        // Note: this only works because every proposition appears in only one disj. LM
-        assert(!simple_landmark_exists(a));
-        assert(disj_lms_to_nodes.find(a) != disj_lms_to_nodes.end());
-        return *(disj_lms_to_nodes.find(a)->second);
+    int get_num_disjunctive_landmarks() const {
+        return num_disjunctive_landmarks;
     }
-
-    int number_of_disj_landmarks() const {
-        return disj_lms;
+    int get_num_conjunctive_landmarks() const {
+        return num_conjunctive_landmarks;
     }
-    int number_of_conj_landmarks() const {
-        return conj_lms;
-    }
-    int number_of_edges() const;
-
-    bool simple_landmark_exists(const FactPair &lm) const; // not needed by HMLandmark
-    bool disj_landmark_exists(const std::set<FactPair> &lm) const;  // not needed by HMLandmark
-    bool landmark_exists(const FactPair &lm) const; // not needed by HMLandmark
-    bool exact_same_disj_landmark_exists(const std::set<FactPair> &lm) const;
-
-    LandmarkNode &landmark_add_simple(const FactPair &lm);
-    LandmarkNode &landmark_add_disjunctive(const std::set<FactPair> &lm);
-    LandmarkNode &landmark_add_conjunctive(const std::set<FactPair> &lm);
-    void remove_node_if(const std::function<bool (const LandmarkNode &)> &remove_node);
-    LandmarkNode &make_disj_node_simple(const FactPair &lm); // only needed by LandmarkFactorySasp
-    void set_landmark_ids();
-    void dump() const;
-private:
-    int conj_lms;
-    int disj_lms;
+    int get_num_edges() const;
 
-    utils::HashMap<FactPair, LandmarkNode *> simple_lms_to_nodes;
-    utils::HashMap<FactPair, LandmarkNode *> disj_lms_to_nodes;
-    Nodes nodes;
-    const TaskProxy &task_proxy;
-
-    void dump_node(const std::unique_ptr<LandmarkNode> &node) const;
-    void dump_edge(int from, int to, EdgeType edge) const;
 
-    void remove_node_occurrences(LandmarkNode *node);
+    // only needed only by non-landmarkgraph-factories
+    LandmarkNode *get_landmark(int index) const;
+    // only needed only by non-landmarkgraph-factories
+    LandmarkNode *get_landmark(const FactPair &fact) const;
+    // needed only by landmarkgraph-factories.
```

Comment:
This comment refers to all methods below it. I saw you copied that to some of them, but not all. I think it would make sense to keep the previous grouping in some form, e.g., by having an empty line here and modifying the comment to say that all of the below methods are used by landmark (graph) factories.

Reply:
I think the grouping of the methods makes more sense the way it is now and gives a better picture of what the public interface of the class currently is (all getters together, all add methods together, etc.). 
If I understood the comments correctly, you collected where each method was used so we could potentially remove or merge some of them, right? If so, the comments are only temporary and the interface will change anyway. Then I'd rather repeat them because it makes it clearer what they refer to and we can remove individual methods without looking at a larger context.
If the comments were meant as a long-term thing, I don't see much use in them.

Reply:
To more clear, maybe we also should change the comments from "this is used from X" to "This is only necessary for X and can be removed once the functionality is part of Y". Maybe it is easier to discuss this live (on discord or zoom).

Reply:
Yes, the comments are "temporary" since 2012 :) I'm fine with splitting the groups up, but to not lose the information (which Salomé and Clemens found very helpful when starting refactoring), please copy them to **all** public methods. This is currently not the case, there are many methods without a categorizing comment and some only have specific comments saying which factory uses/doesn't use it (but not if this code is used only by factories or other parts). Feel free to rephrase the comments as you like.